scipy/signal/spectral.py

```diff
@@ -511,11 +509,401 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+        segment length. Defaults to None. When specified, the COLA
+        constraint must be met (see Notes below), and should match the
+        parameter used to generate the STFT.
+    centered : bool, optional
+        Specifies whether the input signal was padded via odd extension, by
+        using ``centered=True`` in `stft`. Defaults to True.
+    time_axis : int, optional
+        Where the time segments of the STFT is located; the default is the
+        last axis (i.e. ``axis=-1``).
+    freq_axis : int, optional
+        Where the frequency axis of the STFT is located; the default is the
+        penultimate axis (i.e. ``axis=-2``).
+
+    Returns
+    -------
+    t : ndarray
+        Array of output data times.
+    x : ndarray
+        iSTFT of Zxx.
+
+    See Also
+    --------
+    lombscargle: Lomb-Scargle periodogram for unevenly sampled data
+    welch: Power spectral density by Welch's method.
+    spectrogram: Spectrogram by Welch's method.
+    csd: Cross spectral density by Welch's method.
+    stft: Short Time Fourier Transform
+
+    Notes
+    -----
+    In order to enable inversion of an STFT via the inverse STFT in
+    `istft`, the signal windowing must obey the constraint of "Constant
+    OverLap Add" (COLA). This ensures that every point in the input data is
+    equally weighted, thereby avoiding aliasing and allowing full
+    reconstruction. Whether a choice of `window`, `nperseg`, and `noverlap`
+    satisfy this constraing can be tested with
+    `scipy.signal.spectral.check_COLA`.
+
+    .. versionadded:: 0.18.0
+
+    References
+    ----------
+    .. [1] Oppenheim, Alan V., Ronald W. Schafer, John R. Buck
+           "Discrete-Time Signal Processing", Prentice Hall, 1999.
+    .. [2] Daniel W. Griffin, Jae S. Limdt "Signal Estimation from Modified
+           Short Fourier Transform", IEEE 1984, 10.1109/TASSP.1984.1164317
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+
+    Generate a test signal, a 2 Vrms sine wave whose frequency linearly
+    changes with time from 1kHz to 2kHz, corrupted by 0.001 V**2/Hz of
+    white noise sampled at 1024 Hz.
+
+    >>> fs = 1024
+    >>> N = 10*fs
+    >>> amp = 2 * np.sqrt(2)
+    >>> noise_power = 0.001 * fs / 2
+    >>> time = np.arange(N) / fs
+    >>> freq = np.linspace(1e3, 2e3, N)
+    >>> x = amp * np.sin(2*np.pi*freq*time)
+    >>> x += np.random.normal(scale=np.sqrt(noise_power), size=time.shape)
+
+    Compute the STFT, and recover the original signal via inverse STFT
+
+    >>> f, t, Zxx = signal.stft(x, fs, centered=True)
+    >>> _, xrec = signal.istft(Zxx, fs, centered=True)
+
+    Compare the recovered with the original signal
+
+    >>> plt.plot(time, x, time, xrec.real, time, x-xrec.real)
+    >>> plt.xlim([0, 1])
+    >>> plt.xlabel('Time [sec]')
+    >>> plt.ylabel('Signal')
+    >>> plt.legend(['Original', 'Recovered', 'Difference'])
+    >>> plt.show()
```

Comment:
The plot is not so pretty:

https://circle-artifacts.com/gh/scipy/scipy/194/artifacts/0/home/ubuntu/scipy/doc/build/html-scipyorg/generated/scipy.signal.istft.html#scipy.signal.istft

Maybe a cooler input signal would be better? If not, at least adjusting the Y limits would be good.